src/Compilers/CSharp/Test/Semantic/Semantics/SwitchTests.cs

```diff
@@ -2918,6 +2918,35 @@ public static void Main()
                 );
         }
 
+        [Fact]
+        [WorkItem(33783, "https://github.com/dotnet/roslyn/issues/33783")]
+        public void UnreachableDefaultInBoolSwitch()
+        {
+            var text = @"
+public class TestClass
+{
+    public static void Main()
+    {
+        bool b = false;
+        switch (b)
+        {
+            case true:
+                break;
+            case false:
+                break;
+            default:
```

Comment:
I'd add tests for `default when ...` and `default` in a switch statement with type tests

Reply:
There are no backward compat issues for a switch in which a when clause or type tests appears.  We have tests for those scenarios.

`default when ...` is a syntax error.